virt/kvm/nitro_main.c

```diff
@@ -133,3 +136,58 @@ int nitro_ioctl_continue(struct kvm_vcpu *vcpu){
 int nitro_is_trap_set(struct kvm *kvm, uint32_t trap){
   return kvm->nitro.traps & trap;
 }
+
+int nitro_add_syscall_filter(struct kvm *kvm, uint64_t syscall_nb)
+{
+
+	// search if already present
+	int i;
+	for (i = 0; i < kvm->nitro.syscall_filter_size; i++)
+		if (kvm->nitro.syscall_filter[i] == syscall_nb)
+			return 0;
+
+	// insert
+	kvm->nitro.syscall_filter_size++;
+	if (kvm->nitro.syscall_filter_size > NITRO_SYSCALL_FILTER_MAX)
+	{
+		// error, already at max size
+		kvm->nitro.syscall_filter_size--;
+		return 1;
+	}
+	int index = kvm->nitro.syscall_filter_size - 1;
+	kvm->nitro.syscall_filter[index] = syscall_nb;
+	return 0;
+}
```

Comment:
Rather than keeping a counter and an index and increment/decrement them I'd rather store the actual index which is much less error prone.

```
struct nitro
{
    int syscall_filter_index;  // or filter_last ?
    ...
};

int nitro_add_syscall_filter(struct kvm *kvm, uint64_t syscall_nb)
{
    // check the syscall is not already there (I'd rather use a hashtable than an array)

    if (kvm->nitro.syscall_filter_index < NITRO_SYSCALL_FILTER_MAX)
    {
        kvm->nitro.syscall_filter_index++;
        kvm->nitro.syscall_filter[kvm->nitro.syscall_filter_index] = syscall_nb;
    }
    else
    {
        return 1;
    }

    return 0;
}
```

This is more concise, readable and c-like.